src/RA_AchievementSet.cpp

```diff
@@ -388,97 +314,56 @@ BOOL AchievementSet::FetchFromWebBlocking(ra::GameID nGameID)
 }
 
 _Use_decl_annotations_
-BOOL AchievementSet::LoadFromFile(ra::GameID nGameID)
+bool AchievementSet::LoadFromFile(ra::GameID nGameID)
 {
     if (nGameID == 0)
-        return TRUE;
-
-    const auto sFilename{ GetAchievementSetFilename(nGameID) };
-    std::ifstream ifile{ sFilename };
-    if (!ifile.is_open())
-    {
-        //	Cannot open file
-        RA_LOG("Cannot open file %s\n", ra::Narrow(sFilename).c_str());
-        char sErrMsg[2048U]{};
-        strerror_s(sErrMsg, errno);
-        RA_LOG("Error: %s\n", sErrMsg);
-        return FALSE;
-    }
+        return true;
 
-    //	Store this: we are now assuming this is the correct checksum if we have a file for it
-    g_pCurrentGameData->SetGameID(nGameID);
+    auto& pLocalStorage = ra::services::ServiceLocator::GetMutable<ra::services::ILocalStorage>();
 
     if (m_nSetType == Type::Local)
     {
-        auto buffer{ std::make_unique<char[]>(4096U) };
+        auto pData = pLocalStorage.ReadText(ra::services::StorageItemType::UserAchievements, std::to_wstring(static_cast<unsigned int>(nGameID)));
+        if (pData == nullptr)
+            return false;
 
-        //	Get min ver: //	UNUSED at this point? TBD
-        ifile.getline(buffer.get(), 4096LL);
+        std::string sLine;
+        pData->GetLine(sLine); // version used to create the file
+        pData->GetLine(sLine); // game title
 
-        //	Get game title:
-        ifile.getline(buffer.get(), 4096LL);
+        while (pData->GetLine(sLine))
         {
-            const auto nCharsRead{ ifile.gcount() };
-            if (nCharsRead > 0)
+            if (!sLine.empty() && isdigit(sLine[0]))
```

Comment:
Prefer bounds checking whenever possible.
`sLine[0]` could be` sLine.at(0)` or `sLine.front()`.
Sometimes we can't but that'll get solved over time.

Reply:
`empty` handles the bounds checking. 

Why is either of the alternatives better?

Reply:
All I know is the index operator should be avoided unless you have no choice.
If you don't want to change it that's fine as well.

In this case it's not a big deal, but indexing like this can cause loads of security issues in other cases.

Reply:
will change to `front`.